File: medium/numbers/number.kt
fun change(text:String):String{
    
 var dict_numbers = mapOf(Pair("0","zero"),Pair("1","one"),
                      Pair("2","two"),Pair("3","three"),
    Pair("4","four"),Pair("5","five"),Pair("6","six"),
    Pair("7","seven"),Pair("8","eight"),Pair("9","nine"))
 
 	var text_output = text.replace("10","ten")
    
    for((key,value) in dict_numbers){
        if(key in text ){
            text_output = text_output.replace(key,value)
        }
    }
    
    return(text_output)
}	



fun main(args: Array<String>){
	
    println("Text : ")
    var text = readLine()
    
    println(change(text))     
      
}

File: medium/numbers/number.rb
"""
@author	André Peralta
@version 20/05/2021
"""

def replace_numbers text

	dict_numbers = {'0'=>'zero','1'=>'one',
		'2'=>'two','3'=>'three',
		'4'=>'four','5'=>'five','6'=>'six',
		'7'=>'seven','8'=>'eight','9'=>'nine'}

	if text.index("10")
		text.gsub! "10", "ten"
	end

	dict_numbers.each{ |key,value|

		if text.index(key)
			text.gsub! key, value
		end
	}
	return text
	
end

puts "Text: "
text = gets

print replace_numbers(text)
